flake/modules/home-manager/firefox/extensions/default.nix

```diff
@@ -14,16 +14,15 @@
 
   ext = cfg.extensions;
 
-  # TODO: move the mkForceInstalled function to a local library output defined by the flake.
+  # TODO: move these functions to a local library output defined by the flake.
   # Ideally we would do something like self.lib.mkForceInstalled.
+  removeNullAttrs = filterAttrs (_: value: value != null);
   mkForceInstalled = mapAttrs (_: cfg: {installation_mode = "force_installed";} // cfg);
 
-  # HACK: We shouldn't really abuse optionalAttrs like this here, but use mkMerge.
-  # For some reason though, mkMerge fools mkForceInstalled and causes it to think
-  # we are passing a string (what?), so we use optionalAttrs instead.
+  # HACK: Convert this to use mkMerge once the whole project uses the module system.
```

Comment:
the whole project does use the module system, I'm not sure why you changed this comment

Reply:
No, this file is imported and used in an override.

Reply:
It is still a part of the module system, since it is imported by a module and only its result (the addon set) is used.

Reply:
I can explain on Discord if you want.

Reply:
Actually, I'll just do it here. If you look at mkMerge's implementation, it's something like

```nix
mkMerge = contents: { _type = "merge"; inherit contents; }
```
With that in mind, the function that adds the force install attribute runs on _type, which is of type string (this is where you get the error you were referring to in your comment, because you are trying to concat a set with a string), and on contents, which ends up being in the wrong place. This can still be fixed, but the next step where the set is consumed is the ExtensionSettings policy, which, again, has no idea how to deal with _type=merge, so it would serialize it literally, which would result in an incorrect json.